File: back/brain/crowler/creep.py
import sys
import urllib
import urllib.request as ur
import json
import codecs
from bs4 import BeautifulSoup
import re
import requests
import numpy as np

url = "https://play.google.com/store/getreviews"
struct = {
    "reviewType": "0",
    "pageNum": "50",
    "hl": "pt",
    "id": "com.wonder",
    "reviewSortOrder": "0",
    "xhr": "1"
}

def content(page):
    struct["pageNum"] = str(page)
    data = urllib.parse.urlencode(struct).encode("utf-8")
    request = ur.Request(url, data)
    response = ur.urlopen(request)
    jsond = response.read()
    page = json.loads(jsond[6:])
    try:
        review = page[0][2]
        return review
    except IndexError:
        return None

def select(arr):
  _str = arr.split( )
  for r in _str:
    #   if (math.isnan(r)) == False:
      if _str.index(r) == 3:
        num = r
  return int(num)

def configure(arr):
    x = arr.split( )
    y = np.delete(x, len(x)-1)
    z = np.delete(y, len(y)-1)
    new_str = ''
    for r in z:
        new_str = new_str + r + " "
    return new_str

def main():
    page = 0
    reviews = []
    for r in range(0, 1):
    # while True:
        cont = content(page)
        soup = BeautifulSoup(cont, 'html.parser')
        if soup is None:
            page += 1
            break
        else:
            data = list(soup.children)[1]
            if data is None:
                page += 1
                break
            else:
                header = data.find(class_='review-header')
                auxs = header.find(class_='star-rating-non-editable-container')
                username = header.find(class_='author-name').get_text()
                date = header.find(class_='review-date').get_text()
                score = select(auxs['aria-label'])
                auxb = data.find(class_='review-body').get_text()
                body = configure(auxb)
                review = {
                    "author": username,
                    "date": date,
                    "score": score,
                    "content": body
                }
                if review is None:
                    break
                else:
                    reviews.append(review)
                    with open('/home/paulomoraes/Projects/blue/back/dataset/datacrowler.txt', 'w') as r: json.dump(reviews, r, ensure_ascii=False)
                    print("review... ", page+1)
                page += 1

if __name__ == "__main__":
    main()

File: back/brain/crowler/creep.js
var request = require('request');
var cheerio = require('cheerio');
var csv = require('csv-write-stream');
var fs = require('fs');
const uri = 'https://play.google.com/store/apps/details?id=com.wonder';

request(uri, function(error, result, html) {
  if(!error) {
    const $ = cheerio.load(html);
    const content = $('.details-info');
    const app = content.find('.id-app-title').text().trim();
    const category = content.find('.document-subtitle.category').text().trim();
    const developer = content.find('.document-subtitle.primary').text().trim();
    const date = $($("*[itemprop = 'datePublished']").get(0)).text().trim();
    const version = $($("*[itemprop = 'softwareVersion']").get(0)).text().trim();
    const versionandroid = $($("*[itemprop = 'operatingSystems']").get(0)).text().trim();
    const reviewsall = $('.reviews-num').text().trim();
    const appscore = convert($('.score-container, .score').text().trim());
    const reviews = getReviews($);
    const struct = {
        app: app,
        category: category,
        developer: developer,
        date: date,
        version: version,
        downloads: versionandroid,
        qtdreviews: reviewsall,
        score: appscore,
        reviews: reviews
    }
    console.log(app);
    show(reviews)
  } else {
    console.log("Check error: " + error.message);
  }
});

function getReviews ($) {
  const reviews = [];
  const reviewsBox = $('div[class=single-review]');
  reviewsBox.each(function (r) {
    const _id = $(this).find('div[class=review-header]').data('reviewid').trim();
    const username = $(this).find('span[class=author-name]').text().trim();
    const date = $(this).find('span[class=review-date]').text().trim();
    const score = isNum($(this).find('.star-rating-non-editable-container').attr('aria-label').trim());
    const content = $(this).find('.review-body').text().trim();
    const review = { _id, username, date, score, content };
    reviews[r] = review;
  });
  return reviews;
}

function convert(n) {
  let br = n.split(",");
  for (var r = 0; r < br.length; r++) value = br[0] + "." + br[1];
  return parseFloat(value);
}

function isNum(nan) {
  let str = nan.split(" ");
  for (var r = 0; r < str.length; r++) num = str[3];
  return parseInt(num);
}

function show(reviews) {
  for (var r = 0; r < reviews.length; r++) {
    console.log(reviews[r])
  }
}
